/**
 * Stream management messages.
 * @author Krzysztof Trzepla
 * @copyright (C) 2015 ACK CYFRONET AGH
 * @copyright This software is released under the MIT license cited in
 * 'LICENSE.txt'
 */
package one.clproto;

// Requests retransmission of stream messages
message MessageRequest {
    required uint64 stm_id        = 1; // ID of stream requested messages
                                       // belong to
    required uint64 lower_seq_num = 2; // lower bound of sequence numbers of
                                       // requested messages
    required uint64 upper_seq_num = 3; // upper bound of sequence numbers of
                                       // requested messages
}

// Informs about sequence number of last successfully processed stream message
message MessageAcknowledgement {
    required uint64 stm_id  = 1; // ID of stream acknowledged messages
                                 // belong to
    required uint64 seq_num = 2; // sequence number of last successfully
                                 // processed message from the stream
}

// Identifies message as part of the stream
message MessageStream {
    required uint64 stm_id  = 1; // stream ID
    required uint64 seq_num = 2; // sequence number of the message in the stream
    optional bool   eos     = 3; // end of stream flag
}

// Reinitializes each message stream
message MessageStreamReset {
}